moveit_ros/trajectory_cache/src/trajectory_cache.hpp

```diff
@@ -0,0 +1,261 @@
+
+  // Fetches the best cartesian trajectory that fits within the requested tolerances
+  // for start and goal conditions, by some db column.
+  warehouse_ros::MessageWithMetadata<moveit_msgs::msg::RobotTrajectory>::ConstPtr
+  fetch_best_matching_cartesian_trajectory(const moveit::planning_interface::MoveGroupInterface& move_group,
+                                           const std::string& move_group_namespace,
+                                           const moveit_msgs::srv::GetCartesianPath::Request& plan_request,
+                                           double min_fraction, double start_tolerance, double goal_tolerance,
+                                           bool metadata_only = false, const std::string& sort_by = "execution_time_s");
+
+  // Put a cartesian trajectory into the database if it is the best matching
+  // cartesian trajectory seen so far.
+  //
+  // Cartesian trajectories are matched based off their start and goal states.
+  // And are considered "better" if they have a smaller planned execution time
+  // than matching cartesian trajectories.
+  //
+  // Optionally deletes all worse cartesian trajectories by default to prune the
+  // cache.
+  bool put_cartesian_trajectory(const moveit::planning_interface::MoveGroupInterface& move_group,
+                                const std::string& move_group_namespace,
+                                const moveit_msgs::srv::GetCartesianPath::Request& plan_request,
+                                const moveit_msgs::msg::RobotTrajectory& trajectory, double execution_time_s,
+                                double planning_time_s, double fraction, bool delete_worse_trajectories = true);
+
+  // QUERY CONSTRUCTION
+  bool extract_and_append_cartesian_trajectory_start_to_query(
+      warehouse_ros::Query& query, const moveit::planning_interface::MoveGroupInterface& move_group,
+      const moveit_msgs::srv::GetCartesianPath::Request& plan_request, double match_tolerance);
+
+  bool extract_and_append_cartesian_trajectory_goal_to_query(
+      warehouse_ros::Query& query, const moveit::planning_interface::MoveGroupInterface& move_group,
+      const moveit_msgs::srv::GetCartesianPath::Request& plan_request, double match_tolerance);
+
+  // METADATA CONSTRUCTION
+  bool extract_and_append_cartesian_trajectory_start_to_metadata(
+      warehouse_ros::Metadata& metadata, const moveit::planning_interface::MoveGroupInterface& move_group,
+      const moveit_msgs::srv::GetCartesianPath::Request& plan_request);
+
+  bool extract_and_append_cartesian_trajectory_goal_to_metadata(
+      warehouse_ros::Metadata& metadata, const moveit::planning_interface::MoveGroupInterface& move_group,
+      const moveit_msgs::srv::GetCartesianPath::Request& plan_request);
+
+private:
+  rclcpp::Node::SharedPtr node_;
+  warehouse_ros::DatabaseConnection::Ptr db_;
+
+  double exact_match_precision_ = 1e-6;
```

Comment:
const?

Reply:
We assign to it in the init function, so for all intents and purposes, it is a configuration variable.

https://github.com/moveit/moveit2/pull/2838/files#diff-d9125925f4de5f1845a808645c58a9d084434915aa70e0b68d15b87e6acc6c42R124